File: src/main/java/um/haberes/rest/model/AdicionalCursoRango.java
/**
 * 
 */
package um.haberes.rest.model;

import java.io.Serializable;
import java.math.BigDecimal;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;

import lombok.AllArgsConstructor;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

/**
 * @author daniel
 *
 */
@Entity
@EqualsAndHashCode(callSuper = false)
@Getter
@Setter
@ToString
@NoArgsConstructor
@AllArgsConstructor
public class AdicionalCursoRango extends Auditable implements Serializable  {

	private static final long serialVersionUID = 2960349518526001791L;

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long adicionalCursoRangoId;
	
	private Integer horasDesde = 0;
	private Integer horasHasta = 0;
	private BigDecimal porcentaje = BigDecimal.ZERO;
	private Long adicionalCursoTablaId;

}

File: src/main/java/um/haberes/rest/kotlin/model/AdicionalCursoRango.kt
package um.haberes.rest.kotlin.model

import jakarta.persistence.*
import um.haberes.rest.model.Auditable
import java.math.BigDecimal

@Entity
@Table
data class AdicionalCursoRango(

    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    var adicionalCursoRangoId: Long? = null,

    var horasDesde: Int = 0,
    var horasHasta: Int = 0,
    var porcentaje: BigDecimal = BigDecimal.ZERO,
    var adicionalCursoTablaId: Long? = null

) : Auditable()
